clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/CanPatch.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler;
+
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.OperationResult;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials;
+import java.util.HashMap;
+
+public interface CanPatch {
+  KubernetesKind kind();
+
+  default OperationResult patch(KubernetesV2Credentials credentials, String namespace, String name,
+    String mergeStrategy, KubernetesManifest manifest) {
+    credentials.patch(kind(), namespace, name, mergeStrategy, manifest);
+
+    KubernetesManifest patchedManifest = manifest.clone();
+    patchedManifest.putIfAbsent("metadata", new HashMap<String, Object>()); // Hack: Set mandatory field
+    patchedManifest.setNamespace(namespace);
+    patchedManifest.setName(name);
+    patchedManifest.setKind(kind());
+    return new OperationResult().addManifest(patchedManifest);
```

Comment:
I've been thinking about this, and I think we should really put the entire manifest back into the context here. This allows users to snapshot the whole state of the manifest post-patch to keep a healthy audit log. 

Can you check what the last-applied-configuration annotation says on the manifest after `patch` has been run?

Reply:
`patch` does not update the last-applied-configuration. There is a `record` flag that can instead update a `change-cause` annotation:
```
    kubernetes.io/change-cause: |-
      kubectl patch deployment nginx-deployment --patch=spec:
        template:
          spec:
            containers:
            - name: patch-demo-ctr-2
              image: redis --record=true

```
However, running another `patch` will replace this with the new patch content.

Reply:
Let's add a flag to enable this record behavior, and keep in on by default. It at the very least provides insight.

Reply:
Sure. We should also probably add a disclaimer somewhere in the docs that it is the latest patch content, not all of them.

Reply:
Done!